File: test/Automation_python_CodeGen_Quality_RealRepo_python/pipelines/Python_E2E_Existing_Pipeline_All_2/code/python_e2e_existing_pipeline_all_2/graph/FlattenSchema_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from python_e2e_existing_pipeline_all_2.config.ConfigStore import *
from python_e2e_existing_pipeline_all_2.udfs.UDFs import *

def FlattenSchema_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    flt_col = in0.columns
    selectCols = [col("customer_id") if "customer_id" in flt_col else col("customer_id"),                   col("first_name") if "first_name" in flt_col else col("first_name"),                   col("last_name") if "last_name" in flt_col else col("last_name"),                   col("phone") if "phone" in flt_col else col("phone"),                   col("email") if "email" in flt_col else col("email"),                   col("country_code") if "country_code" in flt_col else col("country_code"),                   col("account_open_date") if "account_open_date" in flt_col else col("account_open_date"),                   col("account_flags") if "account_flags" in flt_col else col("account_flags")]

    return in0.select(*selectCols)

File: test/Automation_scala_CodeGen_Quality_RealRepo_scala/pipelines/Scala_E2E_Existing_Pipeline_All_2/code/src/main/scala/io/prophecy/pipelines/scala_e2e_existing_pipeline_all_2/graph/FlattenSchema_1.scala
package io.prophecy.pipelines.scala_e2e_existing_pipeline_all_2.graph

import io.prophecy.libs._
import io.prophecy.pipelines.scala_e2e_existing_pipeline_all_2.udfs.PipelineInitCode._
import io.prophecy.pipelines.scala_e2e_existing_pipeline_all_2.udfs.UDFs._
import io.prophecy.pipelines.scala_e2e_existing_pipeline_all_2.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object FlattenSchema_1 {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.select(
      if (in.columns.contains("customer_id")) col("customer_id")
      else col("customer_id"),
      if (in.columns.contains("first_name")) col("first_name")
      else col("first_name"),
      if (in.columns.contains("last_name")) col("last_name")
      else col("last_name"),
      if (in.columns.contains("phone")) col("phone") else col("phone"),
      if (in.columns.contains("email")) col("email") else col("email"),
      if (in.columns.contains("country_code")) col("country_code")
      else col("country_code"),
      if (in.columns.contains("account_open_date")) col("account_open_date")
      else col("account_open_date"),
      if (in.columns.contains("account_flags")) col("account_flags")
      else col("account_flags")
    )

}
